packages/ui/src/components/va-counter/validateCounter.ts

```diff
@@ -0,0 +1,31 @@
+import { warn } from '../../services/utils'
+
+export const validateCounter = (value: string | number, step: number, min?: number, max?: number) => {
+  const val = Number(value)
+  if (!Number.isNaN(val)) {
```

Comment:
```suggestion
  if (!Number.isNaN(val)) {
    warn('The value is not a number or cannot be reduced to a number.')
    return
  }
```

To decrease `if` nesting level.